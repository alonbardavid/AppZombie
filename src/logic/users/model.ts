import {observable} from "mobx";

export class User {
    @observable
    id:number;
    @observable
    username:string;
    @observable
    name:string;
    @observable
    email:string;
    @observable
    phone:string;
}