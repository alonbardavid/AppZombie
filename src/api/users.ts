import {makeRequest} from "src/api/base";
import Schema from "src/api/endpoint-schema";

export async function getCurrentUserProfile():Promise<Schema.UserProfile>{
    const [res,headers] = await makeRequest("/users/1");
    return res;
}
