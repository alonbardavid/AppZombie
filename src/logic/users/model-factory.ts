import {User} from './model';
import faker from 'faker';
import {consistant} from "src/utils/text-utils";

export function makeUser():User{
    return {
        id:faker.random.number(),
        name:faker.name.findName(),
        email:faker.internet.email(),
        username:faker.internet.userName(),
        phone:faker.phone.phoneNumber()
    }
}
export const getRegularUser = consistant(makeUser);