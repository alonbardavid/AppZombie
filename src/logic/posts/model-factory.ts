import {makeUser} from '../users/model-factory';
import {Post} from './model';
import faker from 'faker';

export function makePost():Post{
    return {
        id:faker.random.number(),
        title:faker.hacker.phrase(),
        body:faker.lorem.text(),
        user:makeUser()
    }
}