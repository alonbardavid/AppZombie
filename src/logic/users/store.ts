import {autorun} from "mobx";
import {User} from "./model";
import {ProfileStore} from "src/logic/profile/store";
import {Api} from 'src/api';

export class UsersStore {
    users = {} as {[id:number]:Partial<User>};

    private profileStore:ProfileStore;
    private api:Api;
    constructor(api,profileStore){
        this.profileStore = profileStore;
        this.api = api;
        autorun(()=>{
            this.profileStore.user && this.updateUser(this.profileStore.user);
        })
    }

    updateUser = (user:User)=>{
        let original = this.users[user.id];
        if (!original){
            original = this.users[user.id] = {id:user.id} as any;
        }
        original.username = user.username;
        original.name = user.name;
        original.phone = user.phone;
        original.email = user.email;
        return original;
    };

    getUser = (userId:number):User=>{
        let user = this.users[userId];
        if (!user){
            user = this.users[userId] = new User();
            user.id = userId;
            this.api.getUser(userId).then(this.updateUser);
        }
        return user as User;
    }
}