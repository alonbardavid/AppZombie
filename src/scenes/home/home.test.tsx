import React from 'react';
import renderer from 'react-test-renderer';
import {HomeScene} from "./index";
import {getRegularUser} from "src/logic/users/model-factory";

test("home with user",()=>{
    const tree = renderer.create(
        <HomeScene user={getRegularUser()}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});