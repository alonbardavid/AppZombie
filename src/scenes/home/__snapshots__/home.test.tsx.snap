// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home with user 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        },
        Object {
          "paddingBottom": 10,
        },
      ]
    }
  >
    Welcome 
    Hilbert Fahey
  </Text>
</View>
`;
